query/src/main/scala/filodb/query/exec/AggrOverRangeVectors.scala

```diff
@@ -934,72 +935,72 @@ class QuantileRowAggregator(q: Double) extends RowAggregator {
   }
 }
 
+/**
+  * Map: Every sample is mapped to map with value as key and 1 as value
+  * ReduceMappedRow: Same as ReduceAggregate
+  * ReduceAggregate: Accumulator maintains a map for values and their frequencies. Reduction happens
+  *                  by deserializing items blob into map and aggregating with frequencies of map of accumulator
+  * Present: Each key from map is placed into distinct RangeVectors. RangeVectorKey is formed by appending the key/value
+  *          to existing labelValues.
+  *
+  */
 class CountValuesRowAggregator(label: String, limit: Int = 1000) extends RowAggregator {
 
-  var arr = new Array[Byte](500)
+  var serializedMap = new Array[Byte](500)
+  val rowMap = debox.Map[Double, Int]()
   class CountValuesHolder(var timestamp: Long = 0L) extends AggregateHolder {
     val frequencyMap = debox.Map[Double, Int]()
     val row =  new CountValuesTransientRow
     def toRowReader: MutableRowReader = {
       val size = frequencyMap.size * 12
-      if (arr.length < size) arr = new Array[Byte](size) // Create bigger array
-      serialize(frequencyMap)
+      if (serializedMap.length < size) serializedMap = new Array[Byte](size) // Create bigger array
+      CountValuesTransientRow.serialize(frequencyMap, serializedMap)
```

Comment:
It appears that `object CountValuesTransientRow` can be renamed to `object CountValuesSerDeser`